pkg/dockerregistry/server/configuration/configuration.go

```diff
@@ -3,18 +3,71 @@ package configuration
 import (
 	"bytes"
 	"errors"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
 	"reflect"
 	"strings"
 	"time"
 
+	log "github.com/Sirupsen/logrus"
 	"gopkg.in/yaml.v2"
 
+	//"github.com/docker/distribution/registry/auth"
 	"github.com/docker/distribution/configuration"
 )
 
+// Environment variables.
+const (
+	// DockerRegistryURLEnvVar is a mandatory environment variable name specifying url of internal docker
+	// registry. All references to pushed images will be prefixed with its value.
+	// DEPRECATED: Use the OPENSHIFT_DEFAULT_REGISTRY instead.
+	DockerRegistryURLEnvVar = "DOCKER_REGISTRY_URL"
+
+	// DockerRegistryURLEnvVarOption is an optional environment that overrides the
+	// DOCKER_REGISTRY_URL.
+	DockerRegistryURLEnvVarOption = "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_DOCKERREGISTRYURL"
```

Comment:
This is the third environment variable to provide the same data. I'd rather not add it, but I see the benefit of consistency here.

Nevertheless, the `OpenShiftDefaultRegistry` should still take precedence. And this should be marked as deprecated as well.

Also, the code evaluating this is not preferring this var over  `DOCKER_REGISTRY_URL`.

Reply:
Why should this be marked as deprecated?  Shouldn't this be the preferred mechanism and both "DOCKER_REGISTRY_URL" and "OPENSHIFT_DEFAULT_REGISTRY" should be deprecated?

(of course that means ansible should be updated too)

Reply:
`OPENSHIFT_DEFAULT_REGISTRY` matches the same variable used to configure the master. The intention was to have the same variable for both registry and master.

From the registry's perspective you're right.

Reply:
well we've made a fine mess :)